ieeeuoft/hackathon-template
# Ieee 269 create table for orders page

## Overview

- Resolves [269](https://ieeeuoft.atlassian.net/browse/IEEE-269)

- Created a table for the orders instead of order cards which also includes status
  - filtering, sorting and searching functionality implemented through mui data grid (note: searching functionality is implemented through filtering)
  - currently displays the following information -- can display more information as well as long as it's a property of the `Order` type (note: tables takes in a prop of type `Orders` so it can be used with Redux later)
    - ID -> id
    - Time Ordered -> created_at
    - Team -> team_code
    - Order Qty -> items.length
    - Status -> status 
  - added new statuses only implemented in the frontend (might be useful, might not be, let me know)
    - Pending -- when the received Ids are false
    - In Progress -- when someone is currently preparing the order
  - table currently displays as many rows that can fit the table size

_I also left some comments in the code, if there were better ways to do certain things to prevent the spaghetti code_

![image](https://github.com/ieeeuoft/hackathon-template/assets/86009011/fa314f6e-59cf-453b-96ca-f3201059ab1b)

## Unit Tests Created

- Unit test just makes sure that the table is rendered, any other suggestions for unit tests would be great

## Steps to QA

- Opening the Orders page, you should see a table generated from mockPendingOrders
- Try sorting based on column properties -- mainly asc and desc times
- Try filtering by status
- Try searching for items, using the column filter
  - ![image](https://github.com/ieeeuoft/hackathon-template/assets/86009011/e6e78afe-69c7-491a-af60-6dbdcbab44e8)
- Also try passing in other data of type `Order[]`
